src/scripts/main.js

```diff
@@ -1,3 +1,262 @@
+    <td>${form.children[1].lastChild.value}</td>
+    <td>${form.children[2].lastChild.value}</td>
+    <td>${form.children[3].lastChild.value}</td>
+    <td>$${Number(
+    form.children[4].lastChild.value).toLocaleString('en-US')}</td>
+  `;
+
+  document.querySelector('tbody').append(newPerson);
+  peopleList.push(newPerson);
+
+  propertyList.push({
+    name: form.children[0].lastChild.value,
+    position: form.children[1].lastChild.value,
+    office: form.children[2].lastChild.value,
+    age: +form.children[3].lastChild.value,
+    salary: +form.children[3].lastChild.value,
+    id: propertyList.length,
+  });
+
+  form.reset();
+  pushNotification('Added to the table', 'success');
+});
+
+let newValue;
+let initialValue;
+
+document.querySelector('tbody').addEventListener('dblclick', (clickEvent) => {
+  const targetText = clickEvent.target.innerText;
+
+  initialValue = clickEvent.target.cellIndex !== 4
+    ? targetText
+    : targetText.replace(/\D/g, '');
+
+  clickEvent.target.classList.add('edit');
+
+  if (clickEvent.target.cellIndex === 2) {
+    clickEvent.target.innerHTML = `
+    <select name="age" value="${targetText}" class="cell-input edit">
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    </select>
+    `;
+
+    [...clickEvent.target.children[0].children].forEach(option => {
+      if (option.value === initialValue) {
+        option.setAttribute('selected', 'true');
+      }
+    });
+  } else if (clickEvent.target.cellIndex > 2) {
+    clickEvent.target.innerHTML = `
+    <input type="number" class="cell-input edit" value="${
+  targetText.replace(/\D/g, '')}">
+    `;
+  } else {
+    clickEvent.target.innerHTML = `
+    <input type="text" class="cell-input edit" value="${
+  targetText}">
+    `;
+  }
+});
+
+function saveEdit(condition) {
+  if (condition) {
+    [...document.querySelectorAll('td')].forEach(el => {
+      if (el.childNodes.length === 3) {
+        const editRowId = el.parentElement.rowIndex - 1;
+        const editCell = el.cellIndex;
+
+        el.classList.remove('edit');
+
+        newValue = el.children[0].value.length
+          ? el.children[0].value
+          : initialValue;
+
+        if (editCell === 0 && newValue.length < 4) {
+          newValue = initialValue;
+
+          pushNotification(
+            'Name must consist of at least 4 characters', 'error');
+        }
+
+        if (editCell === 3 && (newValue < 18 || newValue > 90)) {
+          newValue = initialValue;
+
+          pushNotification('Age must be between 18 and 90', 'error');
+        }
+
+        if (editCell === 4) {
+          newValue = `$${Number(newValue).toLocaleString()}`;
+        }
+
+        switch (el.cellIndex) {
```

Comment:
```suggestion
        switch (el.cellIndex) {
```

good case to use data attributes 😉